internal/integritymonitor/integritymonitor.go

```diff
@@ -0,0 +1,266 @@
+			countHashes++
+		}
+
+		ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+		defer cancel()
+
+		query, args := data.NewHashFileData().PrepareBatchQuery(hashData, dd)
+		err := repositories.ExecQueryTx(ctx, query, args...)
+		if err != nil {
+			errC <- err
+		}
+		doneC <- countHashes
+	}()
+
+	return doneC
+}
+
+func compareHashes(
+	ctx context.Context,
+	log *logrus.Logger,
+	hashC <-chan worker.FileHash,
+	directory string,
+	algName string,
+	deploymentData *models.DeploymentData,
+	errC chan<- error,
+) <-chan int {
+	doneC := make(chan int)
+	go func() {
+		defer close(doneC)
+
+		repository := repositories.NewAppRepository(log, repositories.DB().SQL())
+		expectedHashes, err := repository.GetHashData(
+			directory,
+			algName,
+			deploymentData,
+		)
+		if err != nil {
+			errC <- fmt.Errorf("failed get hash data: %w", err)
+			return
+		}
+
+		//convert hashes to map
+		expectedHashesMap := make(map[string]string)
+		for _, h := range expectedHashes {
+			expectedHashesMap[h.FullFilePath] = h.Hash
+		}
+
+		for v := range hashC {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			if h, ok := expectedHashesMap[v.Path]; ok {
+				if h != v.Hash {
+					errC <- &IntegrityError{Type: ErrTypeFileMismatch, Path: v.Path, Hash: v.Hash}
+					return
+				}
+				delete(expectedHashesMap, v.Path)
+			} else {
+				errC <- &IntegrityError{Type: ErrTypeNewFile, Path: v.Path, Hash: v.Hash}
+				return
+			}
+		}
+		for p, h := range expectedHashesMap {
+			errC <- &IntegrityError{Type: ErrTypeFileDeleted, Path: p, Hash: h}
+			return
+		}
+		doneC <- len(expectedHashes)
+	}()
+	return doneC
+}
+
+func integrityCheckFailed(
+	log *logrus.Logger,
+	err error,
+	alertSender alerts.Sender,
+	kubeClient *services.KubeClient,
+) {
+
+	var msg string
+	var path string
+	var integrityError *IntegrityError
+	if errors.As(err, &integrityError) {
+		path = integrityError.Path
+		switch integrityError.Type {
+		case ErrTypeFileMismatch:
+			msg = IntegrityMessageFileMismatch
+		case ErrTypeNewFile:
+			msg = IntegrityMessageNewFileFound
+		case ErrTypeFileDeleted:
+			msg = IntegrityMessageFileDeleted
+		}
+		log.WithError(err).WithField("path", integrityError.Path).Error("check integrity failed")
+	} else {
+		msg = err.Error()
+		log.WithError(err).Error("check integrity failed")
+	}
+
+	kubeData, err := kubeClient.GetKubeData()
+	if err != nil {
+		log.WithError(err).Error("failed get kubernetes data")
+		return
+	}
+
+	deploymentData, err := kubeClient.GetDataFromDeployment(kubeData)
+	if err != nil {
+		log.WithError(err).Error("failed get kubernetes deployment data")
+		return
+	}
+
+	if alertSender != nil {
+		err := alertSender.Send(alerts.Alert{
+			Time:    time.Now(),
+			Message: fmt.Sprintf("Restart deployment %v", deploymentData.NameDeployment),
+			Reason:  msg,
+			Path:    path,
+		})
+		if err != nil {
+			log.WithError(err).Error("Failed send alert")
+		}
+	}
```

Comment:
If we have the `alertSender` package, we don't need to write the related code here. You may move it to that package, like this:
```go
package alert
...
func NewSender() Sender{}

func (Sender) Send(msg string) error{
  // if not nil {} 
  // if enabled {}
  // prepare alert from msg and metadata 
  //  return Send(alert) 
}
```
..and then just: `err:=alertSender(msg)`

Reply:
I suppose if we will make alert sender as global object it may be changed (in another PR)